django/db/models/expressions.py

```diff
@@ -995,14 +995,15 @@ def set_source_expressions(self, exprs):
     def get_source_expressions(self):
         return [self.expression]
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, **kwargs):
```

Comment:
Yeah I think that's a good idea in general. I was thinking that a general `**kwargs` added to the signature in all cases was probably correct, but I've come around to your thinking. Most uses outside of `Func` expressions just ignore the `**kwargs` parameter and create noise.

So this specific call should be (assuming "extra_context" is a good name):

```
def as_sql(self, compiler, connection, template=None, ordering=None, **extra_context):
    if template is not None:
        extra_context['template'] = template
    if ordering is not None:
        extra_context['ordering'] = ordering
    extra_context['ordering'] = ordering
    place_holders = { .. }
    place_holders.update(extra_context)
   ...
```